mysql/datadog_checks/mysql/statement_samples.py

```diff
@@ -694,26 +731,29 @@ def _explain_statement(self, cursor, statement, schema, obfuscated_statement):
                 schema,
                 obfuscated_statement,
             )
-            return None
+            return None, db_explain_error, e
 
         # Use a cached strategy for the schema, if any, or try each strategy to collect plans
         strategies = list(self._preferred_explain_strategies)
-        cached = self._collection_strategy_cache.get(strategy_cache_key)
-        if cached:
-            strategies.remove(cached)
-            strategies.insert(0, cached)
+        strategy_cache = self._collection_strategy_cache.get(strategy_cache_key)
+        if strategy_cache:
+            strategy, _, _ = strategy_cache
+            strategies.remove(strategy)
+            strategies.insert(0, strategy)
 
+        db_explain_error, err = None, None
         for strategy in strategies:
             try:
                 if not schema and strategy == "PROCEDURE":
+                    db_explain_error = DBExplainError.invalid_schema
```

Comment:
This `not schema and strategy == "PROCEDURE"` scenario isn't really an error. It's expected that if there isn't a default schema then we won't try to collect the execution plan using this strategy. 

Maybe we call the error in this case something different, like `procedure_strategy_requires_default_schema`?